internal/connect/command_offset_update.go

```diff
@@ -0,0 +1,140 @@
+
+	connectv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+func (c *offsetCommand) newUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update a connector's offsets.",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.update,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Config JSON file contains offsets which is to be set for the connector",
+				Code: "config.json = {\n                \"offsets\": [\n                  {\n                    \"partition\": {\n                      \"kafka_partition\": 0,\n                      \"kafka_topic\": \"topic_A\"\n                    },\n                    \"offset\": {\n                      \"kafka_offset\": 1000\n                    }\n                  }\n                ]\n              }",
+			},
+			examples.Example{
+				Text: "Update offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset update lcc-123456 --config-file config.json",
+			},
+			examples.Example{
+				Code: "confluent connect offset update lcc-123456 --config-file config.json --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().String("config-file", "", "JSON file containing connector offsets to set to.")
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagFilename("config-file", "json"))
+	cobra.CheckErr(cmd.MarkFlagRequired("config-file"))
+
+	return cmd
+}
+
+func (c *offsetCommand) update(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
+	}
+
+	request, err := c.getAlterOffsetRequestBody(configFile)
+	if err != nil {
+		return err
+	}
+
+	alterOffsetRequestInfo, err := c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
+	if err != nil {
+		return err
+	}
+
+	offsetStatus := connectv1.ConnectV1AlterOffsetStatus{
+		Request: alterOffsetRequestInfo,
+		Status: connectv1.ConnectV1AlterOffsetStatusStatus{
+			Phase: "PENDING",
+		},
+	}
```

Comment:
Why are we defining this if it will always be overwritten below?

Reply:
in case we dont get the status within timeout, we want to mark it pending.